components/TextFields/src/private/MDCTextInputCommonFundament.m

```diff
@@ -55,6 +56,27 @@
   return [UIColor lightGrayColor];
 }
 
+@implementation MDCTextInputClearButton
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+  CGFloat width = CGRectGetWidth(self.bounds);
+  CGFloat height = CGRectGetHeight(self.bounds);
+  CGFloat verticalInset = MIN(0, -(MDCTextInputClearButtonTouchTargetSize - height) / 2);
+  CGFloat horizontalInset = MIN(0, -(MDCTextInputClearButtonTouchTargetSize - width) / 2);
+  self.hitAreaInsets =
+      UIEdgeInsetsMake(verticalInset, horizontalInset, verticalInset, horizontalInset);
+}
+
+- (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event {
+  if (!UIEdgeInsetsEqualToEdgeInsets(self.hitAreaInsets, UIEdgeInsetsZero)) {
+    return CGRectContainsPoint(
+        UIEdgeInsetsInsetRect(CGRectStandardize(self.bounds), self.hitAreaInsets), point);
+  }
+  return [super pointInside:point withEvent:event];
+}
+@end
+
```

Comment:
Can this be accomplished without relying on hitAreaInsets? E.g. can we set the frame of the view to the correct touchable region?

Reply:
This whole PR might actually be irrelevant with https://github.com/material-components/material-components-ios/pull/8784 haha

Reply:
I rebased after #8784 and am now subclassing MDCButton. I can explore other approaches if the team feels strongly about banning hitAreaInsets across the board. I understand the arguments against having hitAreaInsets as a universal solution for meeting touch target requirements but I think that in this internal and isolated case it works pretty well.

Reply:
I updated the description to reflect the subclassing from MDCButton instead of UIButton

Reply:
Nice, thanks Randall

Reply:
Ah, I just noticed the comment about MDCTextInputClearButtonImageSquareWidthHeight. I'm not sure I fully understand what the ask is. I just tried setting it to a very small value (5) and the touch target was still large. I don't think there's an issue here, but I could be wrong.